pkg/domain/domain_sysvars.go

```diff
@@ -118,8 +120,16 @@ func (do *Domain) getExternalTimestamp(ctx context.Context) (uint64, error) {
 	return do.store.GetOracle().GetExternalTimestamp(ctx)
 }
 
-func (do *Domain) changeSchemaCacheSize(size uint64) {
+func (do *Domain) changeSchemaCacheSize(size uint64) error {
+	err := kv.RunInNewTxn(kv.WithInternalSourceType(context.Background(), kv.InternalTxnDDL), do.store, true, func(_ context.Context, txn kv.Transaction) error {
```

Comment:
```suggestion
	err := kv.RunInNewTxn(kv.WithInternalSourceType(do.ctx, kv.InternalTxnDDL), do.store, true, func(_ context.Context, txn kv.Transaction) error {
```

Maybe add a field `ctx` to `Domain`, and initialize it in `Domain.Init()`.

Reply:
Or move `Domain.Init()` ctx initialization to `NewDomain`.

Reply:
e67b16a use the context pass by setVar